tests/conftest.py

```diff
@@ -1096,15 +1105,14 @@ def _ha_orm_quote(mixed, ident):
 
 @pytest.fixture
 def hass_recorder(
-    recorder_db_url,
-    enable_nightly_purge,
-    enable_statistics,
-    enable_statistics_table_validation,
+    recorder_db_url: str,
+    enable_nightly_purge: bool,
+    enable_statistics: bool,
+    enable_statistics_table_validation: bool,
     hass_storage,
-):
+) -> Generator[Callable[..., HomeAssistant], None, None]:
     """Home Assistant fixture with in-memory recorder."""
-    # Local import to avoid processing recorder and SQLite modules when running a
-    # testcase which does not use the recorder.
```

Comment:
Should we keep these comments to prevent someone from doing a cleanup in the future where the imports are moved to the top?

Reply:
I moved the comment at the top (in `if TYPE_CHECKING`).
I figured that inside each function it was sufficient to have the pylint disable: `# pylint: disable-next=import-outside-toplevel`